generator/generator.dart

```diff
@@ -21,21 +25,97 @@ Future<void> main() async {
   print(_success);
 
   print('\nGenerating $_generatedFilePath...');
-  _generateDartFile(fontDirectory);
+  await _writeDartFile(_generateDartCode(fontDirectory));
   print(_success);
 
   print('\nFormatting $_generatedFilePath...');
   await Process.run('flutter', ['format', _generatedFilePath]);
   print(_success);
 }
 
-const _generatedFilePath = 'lib/google_fonts.dart';
 const _success = 'Success!';
 
-void _generateDartFile(Directory fontDirectory) {
-  final outFile = File(_generatedFilePath);
-  final outFileWriteSink = outFile.openWrite();
+/// Gets the latest font directory. Versioned directories are hosted on the
+/// Google Fonts server. We try to fetch each directory one by one until we hit
+/// the last one. We know we reached the end if requesting the next version
+/// results in a 404 response.
+/// Other types of failure should not occur. For example, if the internet
+/// connection gets lost while downloading the directories, we just crash. But
+/// that's okay for now, because the generator is only executed in trusted
+/// environments by individual developers.
+// TODO(marcelgarus): Exponential search would make a search in O(log n) instead
```

Comment:
I agree with your reasoning here, but I'm not really concerned with this ever becoming a problem. 

The automatic checks are just to make sure we don't miss an update. If the most up-to-date version got to be 101, we could just change the initial version to 100 and it would only check 100, 101 and 102.

Reply:
You're right, that's probably a better way to handle that. I added an `initialVersion` parameter.